app/models/article.rb

```diff
@@ -416,6 +416,17 @@ def access_by?(user)
     user.admin? || user_id == user.id
   end
 
+  def merge_with(other_article_id)
+    Article.find(other_article_id).comments.each do |comment|
+      comment.article_id = self.id
+      comment.save
+    end
+    merge_article = Article.find(other_article_id)
```

Comment:
It seems like this and line 420 are both doing the same `find` operation which means it is calling the database an extra time. Consider how to change this to `find` the article only once.